packages/ra-core/src/actions/dataActions/crudUpdate.ts

```diff
@@ -12,7 +13,8 @@ export const crudUpdate = (
     data: any,
     previousData: any,
     basePath: string,
-    redirectTo: RedirectionSideEffect = 'show'
+    redirectTo: RedirectionSideEffect = 'show',
+    refresh: RefreshSideEffect = true
```

Comment:
Do we need a refresh if we have a redirect ? I think the default value should be `false`

Reply:
same as with delete, with undoable false and redirect to same page, we need refresh to happen if no redirect

Reply:
If you choose to disable optimistic updates and to redirect to the same page, isn't it your responsability to set the refresh to `true`? This will trigger a refresh and a redirection for all default cases

Reply:
I'm surprised it was accepted on delete btw

Reply:
but you cannot set refresh true using provided props,
the only workaround is to create new redux action and set it there

but many ppl will not even notice it, cause they would think that refresh happens like it does if its undoable (with undoable it works with the redirect on same page)

Reply:
I'll consider this a workaround until v3 is out then

Reply:
this "workaround" is already in next branch for delete https://github.com/marmelab/react-admin/blob/ac43cadc9f461c4dff5c02c4f0390f884206c463/packages/ra-core/src/actions/dataActions/crudDelete.ts#L16

Reply:
We'll probably won't need them anymore soon